src/main/java/com/celements/pagetype/service/PageTypeService.java

```diff
@@ -66,13 +66,15 @@
   private List<IPageTypeCategoryRole> pageTypeCategoryList;
 
   private final ConcurrentMap<String, IPageTypeCategoryRole> typeNameToCatCache = new ConcurrentHashMap<>();
+  private final ConcurrentMap<String, IPageTypeCategoryRole> typeNameToCatCacheDeprecated = new ConcurrentHashMap<>();
 
   @Override
   public void resetTypeNameToCatCache() {
     typeNameToCatCache.clear();
+    typeNameToCatCacheDeprecated.clear();
   }
 
-  private Map<String, IPageTypeCategoryRole> getTypeNameToCategoryMap() {
+  private Map<String, IPageTypeCategoryRole> getTypeNameToCategoryMap(boolean deprecated) {
```

Comment:
More Details in Martin Fowler: Refactoring p. 279 (Separate Query from Modifier)